gem/lib/metasploit-payloads/crypto.rb

```diff
@@ -0,0 +1,44 @@
+require 'openssl'
+
+module MetasploitPayloads
+  module Crypto
+    CIPHER_NAME = 'chacha20'.b.freeze
+    IV = 'EncryptedPayload'.b.freeze # 16 bytes
+    KEY = 'Rapid7MetasploitEncryptedPayload'.b.freeze # 32 bytes
+    ENCRYPTED_PAYLOAD_HEADER = 'encrypted_payload_chacha20_v1'.ljust(64, '_').b.freeze
```

Comment:
```suggestion
    ENCRYPTED_PAYLOAD_HEADER_PREFIX = "msf".b.freeze
```

And we'll try to aim for:

```
prefix = ENCRYPTED_PAYLOAD_HEADER_PREFIX + cipher_number_1_byte + key_version_1_byte
```

Where for now:
- chacha20 would be 1
- key version would be 1

In the future if our cipher changes, we can swap to the cipher number to 2. Likewise if the key ever needs to rotate, we can increment the key version number and change the key